pkgs/development/libraries/librime/default.nix

```diff
@@ -20,12 +20,26 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-tflWBfH1+1AFvkq0A6mgsKl+jq6m5c83GA56LWxdnlw=";
   };
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  nativeBuildInputs = [ cmake pkg-config xorgproto utf8cpp ];
 
   buildInputs = [ boost glog leveldb marisa opencc yaml-cpp gtest capnproto ]
               ++ plugins; # for propagated build inputs
 
-  preConfigure = copyPlugins;
+  strictDeps = true;
+
+  postPatch = ''
+    rm --recursive --verbose include/{utf8,utf8.h,X11}
+  '';
+
+
+  preConfigure = ''
+    cmakeFlagsArray=(
+      $cmakeFlagsArray
+      "-DCMAKE_CXX_FLAGS=-I${lib.getDev utf8cpp}/include/utf8cpp -I${lib.getDev xorgproto}/include"
```

Comment:
Is it necessary to set -I for xorgproto?  I do not think other packages do so.

Reply:
Because librime doesn't use pkg-config and cmake to find xorgproto package and never add it into include path.
It checks  `X11/keysym.h` at first, and then add the include path, so we have to add it explictly.

Reply:
It builds fine in my [attempt][1] without adding -I for xorgproto and keysym is indeed found according to the log.
I notice that xorgproto needs to be put in buildInputs when strictDeps is true.

[1]: https://github.com/NixOS/nixpkgs/commit/a250fe47790fad05c5d71cc2e99081df794525b9

Reply:
Because cc-wrapper adds an env hook to add all buildInputs pkgs into include search path, which is unrelated on `strictDeps`. The hook is tiggered in `stdenv` by `findInputs` with condition `(( hostOffset <= targetOffset )) || exit 1`,  why `nativeBuildInputs` doesn't work.